papaya-web/src/main/java/pl/edu/agh/papaya/rest/users/controller/UsersController.java

```diff
@@ -0,0 +1,53 @@
+package pl.edu.agh.papaya.rest.users.controller;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.util.Optional;
+import javax.servlet.http.HttpServletRequest;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.RestController;
+import pl.edu.agh.papaya.api.model.UserDto;
+import pl.edu.agh.papaya.api.service.MeApi;
+import pl.edu.agh.papaya.api.service.UsersApi;
+import pl.edu.agh.papaya.mappers.UserMapper;
+import pl.edu.agh.papaya.model.User;
+import pl.edu.agh.papaya.security.UserContext;
+import pl.edu.agh.papaya.service.user.UserService;
+import pl.edu.agh.papaya.util.ResourceNotFoundException;
+
+@RestController
+@RequiredArgsConstructor
+public class UsersController implements UsersApi, MeApi {
+
+    private final UserService userService;
+
+    private final UserMapper userMapper;
+
+    private final UserContext userContext;
+
+    @Override
+    public ResponseEntity<UserDto> getAuthenticatedUser() {
+        return getUser(userContext.getUserIdn());
+    }
+
+    @Override
+    public ResponseEntity<UserDto> getUser(Long userId) {
+        User user = userService.getById(userId).orElseThrow(ResourceNotFoundException::new);
```

Comment:
A to chyba powinno sprawdzać, czy jesteś adminem projektu, do którego należy ten użytkownik?

Reply:
w sumie trudno powiedzieć, bo z jednej strony nie ma potrzeby, żeby wszyscy to wiedzieli, a z drugiej nie ma tam żadnych _wrażliwych_ danych - a kiedyś może będzie jakaś lista osób w projekcie dostępna dla wszystkich i wtedy się to przyda

Reply:
Email i fejkowe imię i nazwisko nie są wrażliwe???

Reply:
Nie no, ewentualnie ten email

Reply:
Ale nie chce mi się robić ifa

Reply:
Więc przekonałeś mnie w pełni